### ISSUE TITLE
[homematic] Exceptions in the log after updating CCU3 from 3.43.15 to 3.45.7

### ISSUE THREAD

**Issue by sidamos** [openhab/openhab-addons#5725]
openHAB 2.4.0.

After updating my CCU3 from 3.43.15 to 3.45.7, I see these exceptions every few minutes in the openHAB log:
```
2019-06-14 08:48:30.814 [WARN ] [org.eclipse.jetty.server.HttpChannel] - /
java.lang.ClassCastException: java.lang.String cannot be cast to java.util.Map
	at org.eclipse.smarthome.binding.homematic.internal.communicator.server.RpcResponseHandler.handleMethodCall(RpcResponseHandler.java:64) ~[?:?]
	at org.eclipse.smarthome.binding.homematic.internal.communicator.server.XmlRpcServer$ResponseHandler.handle(XmlRpcServer.java:126) ~[?:?]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132) ~[84:org.eclipse.jetty.server:9.4.11.v20180605]
	at org.eclipse.jetty.server.Server.handle(Server.java:531) ~[84:org.eclipse.jetty.server:9.4.11.v20180605]
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:352) [84:org.eclipse.jetty.server:9.4.11.v20180605]
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260) [84:org.eclipse.jetty.server:9.4.11.v20180605]
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:281) [75:org.eclipse.jetty.io:9.4.11.v20180605]
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:102) [75:org.eclipse.jetty.io:9.4.11.v20180605]
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118) [75:org.eclipse.jetty.io:9.4.11.v20180605]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333) [87:org.eclipse.jetty.util:9.4.11.v20180605]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310) [87:org.eclipse.jetty.util:9.4.11.v20180605]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168) [87:org.eclipse.jetty.util:9.4.11.v20180605]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126) [87:org.eclipse.jetty.util:9.4.11.v20180605]
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366) [87:org.eclipse.jetty.util:9.4.11.v20180605]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:762) [87:org.eclipse.jetty.util:9.4.11.v20180605]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:680) [87:org.eclipse.jetty.util:9.4.11.v20180605]
	at java.lang.Thread.run(Thread.java:748) [?:?]
```

```
2019-06-14 08:48:30.816 [ERROR] [nal.communicator.server.XmlRpcServer] - Premature end of file.
org.xml.sax.SAXParseException: Premature end of file.
	at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.createSAXParseException(ErrorHandlerWrapper.java:203) ~[?:?]
	at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.fatalError(ErrorHandlerWrapper.java:177) ~[?:?]
	at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:400) ~[?:?]
	at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:327) ~[?:?]
	at com.sun.org.apache.xerces.internal.impl.XMLScanner.reportFatalError(XMLScanner.java:1472) ~[?:?]
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$PrologDriver.next(XMLDocumentScannerImpl.java:1014) ~[?:?]
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:602) ~[?:?]
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:505) ~[?:?]
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:842) ~[?:?]
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:771) ~[?:?]
	at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141) ~[?:?]
	at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1213) ~[?:?]
	at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:643) ~[?:?]
	at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl.parse(SAXParserImpl.java:327) ~[?:?]
	at org.eclipse.smarthome.binding.homematic.internal.communicator.message.XmlRpcResponse.<init>(XmlRpcResponse.java:52) ~[212:org.eclipse.smarthome.binding.homematic:0.10.0.oh240]
	at org.eclipse.smarthome.binding.homematic.internal.communicator.server.XmlRpcServer$ResponseHandler.handle(XmlRpcServer.java:122) [212:org.eclipse.smarthome.binding.homematic:0.10.0.oh240]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132) [84:org.eclipse.jetty.server:9.4.11.v20180605]
	at org.eclipse.jetty.server.Server.handle(Server.java:531) [84:org.eclipse.jetty.server:9.4.11.v20180605]
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:407) [84:org.eclipse.jetty.server:9.4.11.v20180605]
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260) [84:org.eclipse.jetty.server:9.4.11.v20180605]
	at org.eclipse.jetty.io.AbstractConnect
```

**Comment by XpLoSiON** [comment]
Same issue in my enviroment

Debian Stretch (9.9) 
openjdk version "1.8.0_212"
OpenJDK Runtime Environment (build 1.8.0_212-8u212-b03-2~deb9u1-b03)
OpenJDK 64-Bit Server VM (build 25.212-b03, mixed mode)
OpenHab 2.5.0.M1

Homematic CCU3 with Firmware 3.45.7

I enabled Trace and get this following infos.



```
11:00:56.470 [DEBUG] [communicator.AbstractHomematicGateway] - Received new (Boolean) value 'false' for '000AD8A990CE23:0#SABOTAGE' from gateway with id '3014F711A0001F98A9A585B0'
11:00:56.471 [DEBUG] [communicator.AbstractHomematicGateway] - Received new (Boolean) value 'false' for '000AD8A990CE23:0#CONFIG_PENDING' from gateway with id '3014F711A0001F98A9A585B0'
11:00:56.473 [DEBUG] [communicator.AbstractHomematicGateway] - Received new (Boolean) value 'false' for '000AD8A990CE23:0#DUTY_CYCLE' from gateway with id '3014F711A0001F98A9A585B0'
11:00:56.475 [DEBUG] [communicator.AbstractHomematicGateway] - Received new (Boolean) value 'false' for '000AD8A990CE23:0#LOW_BAT' from gateway with id '3014F711A0001F98A9A585B0'
11:00:56.478 [DEBUG] [communicator.AbstractHomematicGateway] - Received new (Boolean) value 'false' for '000AD8A990CE23:0#UNREACH' from gateway with id '3014F711A0001F98A9A585B0'
11:00:56.480 [DEBUG] [communicator.AbstractHomematicGateway] - Received new (Integer) value '-51' for '000AD8A990CE23:0#RSSI_DEVICE' from gateway with id '3014F711A0001F98A9A585B0'
11:00:56.482 [DEBUG] [communicator.AbstractHomematicGateway] - Received new (Integer) value '0' for '000AD8A990CE23:0#OPERATING_VOLTAGE_STATUS' from gateway with id '3014F711A0001F98A9A585B0'
11:00:56.484 [DEBUG] [communicator.AbstractHomematicGateway] - Received new (Double) value '4.4' for '000AD8A990CE23:0#OPERATING_VOLTAGE' from gateway with id '3014F711A0001F98A9A585B0'
11:00:56.486 [DEBUG] [communicator.AbstractHomematicGateway] - Received new (Integer) value '0' for '000AD8A990CE23:0#ERROR_CODE' from gateway with id '3014F711A0001F98A9A585B0'
11:00:56.489 [TRACE] [rnal.communicator.server.XmlRpcServer] - Server XmlRpcResponse:
<?xml version="1.0" encoding="ISO-8859-1"?>
<methodResponse><params><param><value><array><data><value>event</value></data></array></value></param></params></methodResponse>
11:00:56.504 [TRACE] [rnal.communicator.server.XmlRpcServer] - Server parsed XmlRpcMessage:
system.multicall()
[
        {
                methodName=event
                params=
                [
                        HMIP-3014F711A0001F98A9A585B0
                        000AD8A990CE23:3
                        OPTICAL_ALARM_ACTIVE
                        false
                ]
        }
        {
                methodName=event
                params=
                [
                        HMIP-3014F711A0001F98A9A585B0
                        000AD8A990CE23:3
                        ACOUSTIC_ALARM_ACTIVE
                        false
                ]
        }














]

11:00:56.507 [DEBUG] [communicator.AbstractHomematicGateway] - Received new (Boolean) value 'false' for '000AD8A990CE23:3#OPTICAL_ALARM_ACTIVE' from gateway with id '3014F711A0001F98A9A585B0'
11:00:56.510 [DEBUG] [communicator.AbstractHomematicGateway] - Received new (Boolean) value 'false' for '000AD8A990CE23:3#ACOUSTIC_ALARM_ACTIVE' from gateway with id '3014F711A0001F98A9A585B0'
11:00:56.512 [WARN ] [org.eclipse.jetty.server.HttpChannel ] - /
java.lang.ClassCastException: java.lang.String cannot be cast to java.util.Map
        at org.eclipse.smarthome.binding.homematic.internal.communicator.server.RpcResponseHandler.handleMethodCall(RpcResponseHandler.java:64) ~[?:?]
        at org.eclipse.smarthome.binding.homematic.internal.communicator.server.XmlRpcServer$ResponseHandler.handle(XmlRpcServer.java:125) ~[?:?]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132) ~[84:org.eclipse.jetty.server:9.4.12.v20180830]
        at org.eclipse.jetty.server.Server.handle(Server.java:503) ~[84:org.eclipse.jetty.server:9.4.12.v20180830]
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:364) [84:org.eclipse.jetty.server:9.4.12.v20180830]
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260) [84:org.eclipse.jetty.server:9.4.12.v20180830]
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305) [75:org.eclipse.jetty.io:9.4.12.v20180830]
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103) [75:org.eclipse.jetty.io:9.4.12.v20180830]
        at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118) [75:org.eclipse.jetty.io:9.4.12.v20180830]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333) [87:org.eclipse.jetty.util:9.4.12.v20180830]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310) [87:org.eclipse.jetty.util:9.4.12.v20180830]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168) [87:org.eclipse.jetty.util:9.4.12.v20180830]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126) [87:org.eclipse.jetty.util:9.4.12.v20180830]
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366) [87:org.eclipse.jetty.util:9.4.12.v20180830]
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765) [87:org.eclipse.jetty.util:9.4.12.v20180830]
        at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683) [87:org.eclipse.jetty.util:9.4.12.v20180830]
        at java.lang.Thread.run(Thread.java:748) [?:?]
11:00:56.521 [ERROR] [rnal.communicator.server.XmlRpcServer] - Vorzeitiges Dateiende.
org.xml.sax.SAXParseException: Vorzeitiges Dateiende.
        at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.createSAXParseException(ErrorHandlerWrapper.java:203) ~[?:?]
        at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.fatalError(ErrorHandlerWrapper.java:177) ~[?:?]
        at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:400) ~[?:?]
        at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:327) ~[?:?]
        at com.sun.org.apache.xerces.internal.impl.XMLScanner.reportFatalError(XMLScanner.java:1472) ~[?:?]
        at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$PrologDriver.next(XMLDocumentScannerImpl.java:1014) ~[?:?]
        at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:602) ~[?:?]
        at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:505) ~[?:?]
        at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:842) ~[?:?]
        at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:771) ~[?:?]
        at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141) ~[?:?]
        at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1213) ~[?:?]
        at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:643) ~[?:?]
        at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl.parse(SAXParserImpl.java:327) ~[?:?]
        at org.eclipse.smarthome.binding.homematic.internal.communicator.message.XmlRpcResponse.<init>(XmlRpcResponse.java:52) ~[198:org.eclipse.smarthome.binding.homematic:0.11.0.oh250M1]
        at org.eclipse.smarthome.binding.homematic.internal.communicator.server.XmlRpcServer$ResponseHandler.handle(XmlRpcServer.java:121) [198:org.eclipse.smarthome.binding.homematic:0.11.0.oh250M1]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132) [84:org.eclipse.jetty.server:9.4.12.v20180830]
        at org.eclipse.jetty.server.Server.handle(Server.java:503) [84:org.eclipse.jetty.server:9.4.12.v20180830]
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:419) [84:org.eclipse.jetty.server:9.4.12.v20180830]
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260) [84:org.eclipse.jetty.server:9.4.12.v20180830]
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305) [75:org.eclipse.jetty.io:9.4.12.v20180830]
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103) [75:org.eclipse.jetty.io:9.4.12.v20180830]
        at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118) [75:org.eclipse.jetty.io:9.4.12.v20180830]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333) [87:org.eclipse.jetty.util:9.4.12.v20180830]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310) [87:org.eclipse.jetty.util:9.4.12.v20180830]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168) [87:org.eclipse.jetty.util:9.4.12.v20180830]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126) [87:org.eclipse.jetty.util:9.4.12.v20180830]
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366) [87:org.eclipse.jetty.util:9.4.12.v20180830]
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765) [87:org.eclipse.jetty.util:9.4.12.v20180830]
        at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683) [87:org.eclipse.jetty.util:9.4.12.v20180830]
        at java.lang.Thread.run(Thread.java:748) [?:?]
```

**Comment by steilerDev** [comment]
I have the same issue in my environment (I am using the latest openHAB docker image & CCU Version 3.47.15 [latest pivCCU]).

However I also see issues with my roller shutters (HmIP-FROLL) where the "UP" command does no longer work and also my Duty Cycle for the first time since having this setup is up significantly.

**Comment by steilerDev** [comment]
I have checked the [Changelog](https://www.eq-3.de/downloads/software/firmware/ccu3-firmware/CCU3-Changelog.3.45.7.pdf) for the CCU3 and unfortunately there is no obvious change to the API, however there are certain security patches and "A fix to UPnP", maybe someone of the devs knows where this issue arises from.

**Comment by Owlbertz** [comment]
I am facing the same issue and it seems that the whole Homematic binding stopped working.
The OpenHAB log outputs that it cannot find the device I send a command to at the binding `ccu`.
Worked perfectly fine before the update.

**Comment by sidamos** [comment]
The binding is still working for me in openHAB 2.5.0, but the exceptions also still occur.

**Comment by MHerbst** [comment]
I have never seen these exceptions in my environment. I am using Raspberrymatic which is very similar to the current CCU3 version. What is your openHAB environment?

**Comment by sidamos** [comment]
What do you want to know? ;-) I have those exception with openHAB 2.4.0 and 2.5.0.

**Comment by MHerbst** [comment]
Which operating system, java version are you using? 

If you can easily reproduce this exception it would be fine if you could enable the TRACE mode for the Homematic binding so that we can see the message that causes the exception

**Comment by sidamos** [comment]
Linux, Java 8, 64 Bit. I am attaching the TRACE log.
[openhab.log](https://github.com/openhab/openhab-addons/files/3995635/openhab.log)

**Comment by MHerbst** [comment]
The returned xml message looks OK. I have to look into code and maybe add some additional trace coding

**Comment by smurex** [comment]
Any update on this issue? I have the same problem.

**Comment by sidamos** [comment]
Seems to be fixed in openHAB 2.5.2.

**Comment by MHerbst** [comment]
@sidamos can you please close this issue if the problem no longer exists.
